### ISSUE TITLE
Implement `--tsconfig` param

### ISSUE THREAD

**Issue by fvictorio** [nomiclabs/hardhat#1475]
We have a reserved `--tsconfig` param:

https://github.com/nomiclabs/hardhat/blob/677099d520a3756929145ff450dfd7d39babf036/packages/hardhat-core/src/internal/core/params/hardhat-params.ts#L78

But it's not implemented yet. This will need some research. For example, is it as easy as properly setting the `TS_NODE_PROJECT` variable or do we need to pass it to every part where `ts-node` is registered?

**Comment by demisx** [comment]
We really need this feature. We need to be able to point to a different tsconfig file (i.e. tsconfig.app.json) for compilation. If you have any idea on how this can be currently done I'd highly appreciate the advice.

**Comment by fvictorio** [comment]
@demisx does the `TS_NODE_PROJECT` work? That is, try something like:

```
TS_NODE_PROJECT=/path/to/my/tsconfig.app.json npx hardhat compile
```

**Comment by demisx** [comment]
@fvictorio This worked. Thank you 🙏🏻

**Comment by Rishabhraghwendra18** [comment]
is this issue still open ?

**Comment by alcuadrado** [comment]
It should be closed. Thanks. We'll release it this week.

**Comment by fvictorio** [comment]
Released in `hardhat@2.7.0`. Check the [release notes](https://github.com/nomiclabs/hardhat/releases/tag/hardhat%402.7.0) to learn more.
